state/indexer/block/kv/kv.go

```diff
@@ -86,6 +90,20 @@ func (idx *BlockerIndexer) Index(bh types.EventDataNewBlockEvents) error {
 	return batch.WriteSync()
 }
 
+func (idx *BlockerIndexer) Prune(lastRetainHeight int64, retainHeight int64) {
+	for height := lastRetainHeight; height < retainHeight; height++ {
+		key, err := heightKey(height)
+		if err != nil {
```

Comment:
The error here can be `ErrNotFound` which means the key does not exist, so I am not 100% sure you should panic here. I imagine we might have a block without events for a height.  The deletion errors are fine because they will not happen if the key does not exist.

Reply:
To my understanding, this is not the search for the key, but the creation of one as a function of height. This does not call the BD. I might have misunderstood you though

Reply:
ah right ,I understood it was trying to retrieve the key before deleting. All good, disregard the comment.